src/equation_of_state/MOM_EOS.F90

```diff
@@ -193,6 +202,43 @@ subroutine calculate_density_scalar(T, S, pressure, rho, EOS, rho_ref, scale)
 
 end subroutine calculate_density_scalar
 
+!> Calls the appropriate subroutine to calculate density of sea water for scalar inputs
```

Comment:
I think that perhaps calculate_stanley_density_scalar should follow the same pattern with respect to the dimensions of its arguments and rescaling via the contents of EOS as does calculate_density_scalar, and not the pattern of calculate_density_array, as it currently does.  This is especially true if the calculate_density_array routines are eventually going to be hidden from public exposure.

Reply:
The problem actually goes deeper because second derivatives are always scaled so it is hard to build these Stanley version out of the existing functions. The existing `calculate_density_1d()` calls `calculate_density_array()` and to do so has to "unscale" rho_ref. This did not work for the second derivatives so I've had to write out longer versions of these functions. Addressed in commit e8adc48a36dfe4b12459651ee2893b6670c0e2a1.